src/pdfkit.cs

```diff
@@ -378,35 +1053,98 @@ interface PdfDestination : NSCopying {
 
 		[Export ("point")]
 		CGPoint Point { get; }
-		
+
+		[Mac (10, 7)]
+		[Export ("zoom")]
+		nfloat Zoom { get; set; }
+
 		//Should Compare be more more .Net ified ?
 		[Export ("compare:")]
 		NSComparisonResult Compare (PdfDestination destination);
 	}
 
 	//Add attributes for delegates/events
-	[BaseType (typeof (NSObject), Name="PDFDocument", Delegates=new string [] { "WeakDelegate" }, Events=new Type [] { typeof (PdfDocumentDelegate)})]
+	[iOS (11,0)]
+	[BaseType (typeof (NSObject), Name = "PDFDocument", Delegates = new string [] { "WeakDelegate" }, Events = new Type [] { typeof (PdfDocumentDelegate) })]
 	interface PdfDocument : NSCopying {
+
+		[Field ("PDFDocumentDidUnlockNotification", "+PDFKit")]
+		[Notification]
+		NSString DidUnlockNotification { get; }
+
+		[Field ("PDFDocumentDidBeginFindNotification", "+PDFKit")]
+		[Notification]
+		NSString DidBeginFindNotification { get; }
+
+		[Field ("PDFDocumentDidEndFindNotification", "+PDFKit")]
+		[Notification]
+		NSString DidEndFindNotification { get; }
+
+		[Field ("PDFDocumentDidBeginPageFindNotification", "+PDFKit")]
+		[Notification]
+		NSString DidBeginPageFindNotification { get; }
+
+		[Field ("PDFDocumentDidEndPageFindNotification", "+PDFKit")]
+		[Notification]
+		NSString DidEndPageFindNotification { get; }
+
+		[Field ("PDFDocumentDidFindMatchNotification", "+PDFKit")]
+		[Notification]
+		NSString DidFindMatchNotification { get; }
+
+		[Field ("PDFDocumentDidBeginWriteNotification", "+PDFKit")]
+		[Notification]
+		NSString DidBeginWriteNotification { get; }
+
+		[Field ("PDFDocumentDidEndWriteNotification", "+PDFKit")]
+		[Notification]
+		NSString DidEndWriteNotification { get; }
+
+		[Field ("PDFDocumentDidBeginPageWriteNotification", "+PDFKit")]
+		[Notification]
+		NSString DidBeginPageWriteNotification { get; }
+
+		[Field ("PDFDocumentDidEndPageWriteNotification", "+PDFKit")]
+		[Notification]
+		NSString DidEndPageWriteNotification { get; }
+
 		[Export ("initWithURL:")]
+		[DesignatedInitializer]
 		IntPtr Constructor (NSUrl url);
 
 		[Export ("initWithData:")]
-		IntPtr Constructor  (NSData data);
+		[DesignatedInitializer]
+		IntPtr Constructor (NSData data);
 
 		[Export ("documentURL")]
 		NSUrl DocumentUrl { get; }
 
-		//[Export ("documentRef")]
-		//CGPdfDocumentRef DocumentRef { get; }
+		[Export ("documentRef")]
+		CGPDFDocument Document { get; }
 
+		[Advice ("Use '[Get|Set]DocumentAttributes' instead.")]
```

Comment:
Maybe good to explain in the advice why people should use the '[Get|Set]DocumentAttributes' version. So: "Use the strongly typed '[Get|Set]DocumentAttributes' instead."